execution/evm/execution.go

```diff
@@ -355,80 +383,96 @@ func (c *EngineClient) ExecuteTxs(ctx context.Context, txs [][]byte, blockHeight
 
 			return fmt.Errorf("returned nil PayloadID - (status: %s, latestValidHash: %s)",
 				forkchoiceResult.PayloadStatus.Status,
-				forkchoiceResult.PayloadStatus.LatestValidHash.Hex())
+				latestValidHashHex(forkchoiceResult.PayloadStatus.LatestValidHash))
 		}
 
-		payloadID = forkchoiceResult.PayloadID
+		newPayloadID = forkchoiceResult.PayloadID
 		return nil
 	}, MaxPayloadStatusRetries, InitialRetryBackoff, "ExecuteTxs forkchoice")
 	if err != nil {
 		return nil, 0, err
 	}
 
-	// get payload
-	var payloadResult engine.ExecutionPayloadEnvelope
-	err = c.engineClient.CallContext(ctx, &payloadResult, "engine_getPayloadV4", *payloadID)
-	if err != nil {
-		return nil, 0, fmt.Errorf("get payload failed: %w", err)
+	// Save ExecMeta with payloadID for crash recovery (Stage="started")
```

Comment:
let's remove this nil check and assume store is always set. it is in the constructor.